lib/spike-detector.js

```diff
@@ -0,0 +1,110 @@
+var moment = require('moment');
+var _ = require('lodash');
+var debug = require('debug')('spikeD');
+
+var events = require('events');
+var eventEmitter;
+
+var Deviation = require('./deviation')
+var deviation;
+
+module.exports = function (options) {
+
+  function SpikeD() {
+    events.EventEmitter.call(this);
+    deviation = new Deviation();
+    this.normalTraffic = {};
+    for(var day = 0; day <= options.days; day++){
+      this.normalTraffic[moment().subtract(day, 'days').format('YYMMDD')]= new Array(24);
+    }
+  }
+
+  SpikeD.prototype.__proto__ = events.EventEmitter.prototype;
+
+  SpikeD.prototype.logLastHourCount = function (date) {
+    var previousHourDate = date.subtract(1, 'hour');
+    debug('last hour count was' + ' ' + this.normalTraffic[previousHourDate.format('YYMMDD')][previousHourDate.format('H')]);
+    return this.normalTraffic[previousHourDate.format('YYMMDD')][previousHourDate.format('H')];
+  };
+
+  //TODO: make "spike factor" configurable
+  SpikeD.prototype.checkForSpike = function (date) {
+    if(!moment.isMoment(date)){
+      date = moment(date);
+    }
+    var previousHour = date.subtract(1, 'hour').format('H');
```

Comment:
should this be `previous` or `current` hour? 

so if it is 3:30pm, is the current hour `3` ?